src/os/bluestore/NVMEDevice.cc

```diff
@@ -486,6 +487,17 @@ class NVMEManager {
 
  public:
   NVMEManager() {}
+  ~NVMEManager() {
+    if (!init)
```

Comment:
we still need to wake up the `try_get()` waiters, so i think we should 

1. remove the member variable of `init`, and instead check if the `dpdk_thread` is joinable, before executing `dpdk_thread`, and join it in the dtor only if it's joinable.
1. set all pending `ctx->done` to `true`, before notifying all `try_get()` waiters in `dpdk_thread`.
1. might want to rename `should_probe` to `stopping`.

Reply:
Thanks Kefu.
For point 2, I will promote ctx as NVMEManager member variable then the destroyer can access it.

I will update the change soon. Thanks a lot!

Reply:
i don't follow you. `ctx`s are elements of `probe_queue`. so i think we already have access to them? also, i'd recommend set them in `dpdk_thread` before it exits. it's more readable this way, as we notify the waiter in a single place.

Reply:
Got it. I mentioned another ctx. I will update code.

Thanks a lot! :)